app/controllers/project2.php

```diff
@@ -0,0 +1,279 @@
+												'total_weight',
+												'completedCount',
+												'members',
+												'pTitle'
+											]));
+	}
+
+	/**
+	 * Show the form for editing the specified resource.
+	 * GET /projects/{id}/edit
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function edit($id)
+	{
+		$project 	= 	Project::find($id);
+
+		if($project->isOwner() == false){
+			return Redirect::to('/');
+		}
+
+		$pTitle 		=	"Edit " . $project->name;
+		$total_weight	=	$project->tasks()->where('state','incomplete')->sum('weight');
+		$owner_id		=	$project->user_id;
+		$members 		= 	$project->members()->get();
+
+		return View::make('projects.edit', compact(['project','pTitle','owner_id','total_weight','members']));
+	}
+
+	/**
+	 * Update the specified resource in storage.
+	 * PUT /projects/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function update($id)
+	{
+		$project 	= Project::find($id);
+
+       // Validation
+        $validator = Validator::make(
+            array(
+            	'name' 				=>	Input::get("name"),
+            	'github'			=>	Input::get("github"),
+            	'production'		=>	Input::get("production"),
+            	'stage'				=>	Input::get("stage"),
+            	'dev'				=>	Input::get("dev")
+            	),
+            array(
+            	'name' 				=> 'required',
+            	'github'			=> 'url',
+            	'production'		=> 'url',
+            	'stage'				=> 'url',
+            	'dev'				=> 'url'
+            	)
+        );
+
+        if ($validator->fails())
+        {
+            return Redirect::back()->withErrors($validator)->withInput();
+        }
+
+		$project->name 			= Input::get('name');
+		$project->github 		= Input::get('github');
+		$project->production 	= Input::get('production');
+		$project->stage 		= Input::get('stage');
+		$project->dev 			= Input::get('dev');
+		$project->save();
+
+		return Redirect::back()->with('success', "The project " .Input::get('name'). " has been updated.");
+	}
+
+	/**
+	 * Remove the specified resource from storage.
+	 * DELETE /projects/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function destroy($id)
+	{
+		$pTitle		=	"Projects";
+
+		$project 	= 	Project::find(Input::get("id"));
+
+		// delete everything associated with project
+		Task::where('project_id',Input::get("id"))->delete();		
+		Credential::where('project_id',Input::get("id"))->delete();
+		$project->members()->detach();
+
+
+		// delete the project
+		$project->delete();
+		
+		$counter 	=	0;
+		$user 		=	User::find(Auth::id());
+		$projects 	=	$user->projects()->get();
+		$inProjects =  $user->inProjects()->orderBy('created_at', 'desc')->take(5)->get();		
+								
+		return View::make('projects.index',compact(['projects','counter','inProjects','pTitle']));		
+	}
+
+	/**
+	 * Invites a user to the given project.
+	 * @return Redirect
+	 */
+	public function invite($id){
+		// Validation
+		$rules = ['email' => 'required|email|exists:users,email'];
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "email" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.137:9000/coding_rules#rule_key=php%3AS1192)